google/google-how-to/gce/set-hostname-using-cloudinit.rst

```diff
@@ -23,42 +32,53 @@ Here INSTANCE_NAME is your VM name and the metadata is specified using a KEY=VAL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Due to the way the underlying ``hostname`` command works, whenever a user or tool (such as cloud-init) tries to set the hostname on a system without ``/etc/hostname``, it will only be set transiently and will be overwritten by Google's DHCP service. To avoid this, you'll need to set ``create_hostname_file`` to true in the user-data file:
 
 .. code::
 
     hostname: test123
     create_hostname_file: true
 
-By setting ``create_hostname_file`` to true, you ensure that cloud-init will create the ``/etc/hostname`` file on boot if it does not already exist; and that the hostname will be statically set to the one specified in the user-data file.
+By setting ``create_hostname_file`` to true, you ensure two things: 
 
+#. cloud-init will create the ``/etc/hostname`` file on boot (if it does not already exist) 
+#. hostname will be statically set to the one specified in the user-data file
 
-Option 2: Use DHCP (Google's preferred method)
-----------------------------------------------
 
-Google's preferred method of setting hostname is through its DHCP service. The hostname is set to an automatically generated internal DNS name by default. 
+Creating consistent server farms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you want your own custom name, you can follow the instructions given in `Create a VM instance with a custom hostname`_. In this case, the DHCP service will additionally provide the custom name and will prioritise it to be the default hostname. However, the custom name needs to be a fully qualified DNS name, e.g. something like test123.test.com, and you also need to create a corresponding DNS record for it. 
 
-If you need the hostname to be in a different format, you can use cloud-init for setting it, as described in the :ref:`previous section <use_cloud-init>`.
+Another scenario where this new default can create inconsistencies is in the case of a server farm with images spanning the Ubuntu 24.04 boundary (i.e. both 24.04+ and 23.10-). In this case, if you want a consistent file system layout and hostname style across all images, then you'll have to either remove the ``/etc/hostname`` file from the earlier versions or add it to the later versions.
 
 
-Additional change needed for Ubuntu 23.10 and earlier
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Remove ``/etc/hostname`` from Ubuntu 23.10 and earlier
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-If you want to ensure that GCE images that use Ubuntu 23.10 (or earlier) follow the new GCE default of only setting hostnames through DHCP, then the existing ``/etc/hostname`` must be removed and the cloud-init key ``create_hostname_file`` must be set to false. So your user-data file will now need:  
+Set the cloud-init key ``create_hostname_file`` to false and ensure that ``/etc/hostname`` is deleted during or after first boot. So the user-data file will need:  
 
 .. code::
 
     create_hostname_file: false
```

Comment:
Should we include the hostname in these same userdata files?  E.g.,

```
#cloud-config

hostname: test123.com
create_hostname_file: false
```

Reply:
Good point, yeah we should just have `create_hostname_file` in this one and assume they're setting the hostname some other way